### PR TITLE
update docs & fix CI

### PR DESCRIPTION
**Pull request by dpaiton**
# Resolved Issues
No issues

# Description
- add some minor docstring fixes & moves code to test a PR in the new repo
- fix CI to run lint & tests

### CODE REVIEW

File: crates/hyperdrive-wrappers/build.rs [hunk 1]
```diff
@@ -58,6 +58,12 @@ const TARGETS: &[&str] = &[
 
 fn get_artifacts(artifacts_path: &Path) -> Result<Vec<(String, String)>> {
     let mut artifacts = Vec::new();
+    if !artifacts_path.exists() {
```
Comment (jalextowle) [inline crates/hyperdrive-wrappers/build.rs:61]: We could just build it here with mkdirp. It’s easy enough that I think it’s worth doing

**Review by jalextowle — APPROVED** [review]
LGTM, but we should do a mkdirp

File: crates/hyperdrive-wrappers/build.rs (hunk 1 above)
Comment (dpaiton) [reply crates/hyperdrive-wrappers/build.rs:61]: oh ok cool, it just needs to be an empty dir? or do we need to populate it with something?

File: .github/workflows/rust_lint.yml [hunk 2]
```diff
@@ -20,18 +20,22 @@ jobs:
...
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
 
       - name: check out hyperdrive
-        uses: actions/checkout@main
+        uses: actions/checkout@v3
         with:
           repository: delvtech/hyperdrive
-          ref: "latest"
```
Comment (ryangoree) [inline .github/workflows/rust_lint.yml:28]: What happens if there's no ref? Should we be targeting a specific version?

**Review by ryangoree — APPROVED** [review]

File: .github/workflows/rust_lint.yml (hunk 2 above)
Comment (dpaiton) [reply .github/workflows/rust_lint.yml:28]: good catch; I pushed up a fix for this to grab use the latest tag.

**Pull request merged**